pkg/expr/expr.go

```diff
@@ -390,6 +390,46 @@ func Parse(src string) (ex *Expression, err error) {
 	return ex, nil
 }
 
+// ExtractEQMatches extracts equality sub expressions from the match expression.
+// It only extracts `attribute == literal` type equality matches.
+// It returns a list of  <attribute name, value> such that
+// if **any** of these comparisons is false, the expression will evaluate to false.
+// These sub expressions can be hoisted out of the match clause and evaluated separately.
+// For example
+// destination.service == "abc"  -- Used to index rules by destination service.
+// context.protocol == "tcp"  -- Used to filter rules by context
+func ExtractEQMatches(src string) (map[string]interface{}, error) {
+	ex, err := Parse(src)
+	if err != nil {
+		return nil, err
+	}
+	eqMap := make(map[string]interface{})
+	extractEQMatches(ex, eqMap)
+	return eqMap, nil
+}
+
+func recordEQ(fn *Function, eqMap map[string]interface{}) {
+	if fn.Name == "EQ" && fn.Args[0].Var != nil && fn.Args[1].Const != nil {
```

Comment:
nit: shouldn't it accept yoda-conditions (i.e. `"abc" == destination.service` style)?

Reply:
good point, added comparison and tests.